tests/storage_tests.cpp

```diff
@@ -4,6 +4,26 @@
 
 using namespace sqlite_orm;
 
+#ifdef SQLITE_ORM_CTAD_SUPPORTED
+TEST_CASE("connection control") {
+    const auto openForever = GENERATE(false, true);
+    SECTION("") {
+        SECTION("empty") {
+            auto storage = make_storage("", connection_control{openForever}, on_open([](sqlite3*) {}));
```

Comment:
probably we may add also `on_open` call counter increment inside `on_open` and checking that is was or wasn't called